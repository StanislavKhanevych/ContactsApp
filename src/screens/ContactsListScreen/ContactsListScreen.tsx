import {View, Text} from 'react-native';

type TContactsListScreenProps = {};

const ContactsListScreen = ({}: TContactsListScreenProps) => {
  return (
    <View>
      <Text>Contacts List Screen</Text>
    </View>
  );
};

export default ContactsListScreen;
