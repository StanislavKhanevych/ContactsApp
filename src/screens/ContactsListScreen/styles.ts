import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {flex: 1},
  icon: {
    bottom: 20,
    right: 20,
    position: 'absolute',
    zIndex: 1,
  },
});
