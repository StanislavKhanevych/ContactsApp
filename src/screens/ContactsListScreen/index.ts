import ContactsListScreen from './ContactsListScreen';

export default ContactsListScreen;
