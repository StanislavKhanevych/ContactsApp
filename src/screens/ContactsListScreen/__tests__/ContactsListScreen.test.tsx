import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
import ContactsListScreen from '../../ContactsListScreen/ContactsListScreen';
import {NavigationContainer} from '@react-navigation/native';

jest.mock('../../../services/useContacts', () => ({
  __esModule: true,
  default: jest.fn().mockReturnValue({
    contacts: [
      {
        recordID: '6b2237ee0df85980',
        emailAddresses: [
          {
            label: 'work',
            email: 'carl-jung@example.com',
          },
        ],
        familyName: 'Jung',
        givenName: 'Carl',
        phoneNumbers: [
          {
            label: 'mobile',
            number: '(555) 555-5555',
          },
        ],
        hasThumbnail: true,
        thumbnailPath: 'content://com.android.contacts/display_photo/3',
      },
    ],
    permissionGranted: true,
    setContacts: jest.fn(),
  }),
}));

describe('ContactsListScreen', () => {
  const mockNavigation = {
    navigate: jest.fn(),
    goBack: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders items correctly', () => {
    const {getByTestId} = render(
      <NavigationContainer>
        <ContactsListScreen navigation={mockNavigation} />
      </NavigationContainer>,
    );
    const list = getByTestId('flat-list');

    expect(list).toBeDefined();
  });

  it('checks if search works correctly', async () => {
    const {getByTestId, queryByText} = render(
      <NavigationContainer>
        <ContactsListScreen navigation={mockNavigation} />
      </NavigationContainer>,
    );

    const searchInput = getByTestId('search-bar');
    fireEvent.changeText(searchInput, 'Jung');

    await waitFor(() => {
      expect(queryByText('Jack')).toBeNull();
      expect(queryByText('Jung')).toBeTruthy();
      expect(queryByText('Alice')).toBeNull();
    });
  });

  it('navigates to CreateContact screen on add-contact press', () => {
    const {getByTestId} = render(
      <NavigationContainer>
        <ContactsListScreen navigation={mockNavigation} />
      </NavigationContainer>,
    );
    const addButton = getByTestId('add-contact');
    fireEvent.press(addButton);

    expect(mockNavigation.navigate).toHaveBeenCalledWith('CreateContact');
  });

  it('navigates to ContactDetails screen on contact item press', () => {
    const {getByText} = render(
      <NavigationContainer>
        <ContactsListScreen navigation={mockNavigation} />
      </NavigationContainer>,
    );
    const contactItem = getByText('Jung');
    fireEvent.press(contactItem);

    expect(mockNavigation.navigate).toHaveBeenCalledWith('ContactDetails', {
      contactInfo: {id: '6b2237ee0df85980'},
    });
  });

  it('should match the snapshot', () => {
    const component = render(
      <NavigationContainer>
        <ContactsListScreen navigation={mockNavigation} />
      </NavigationContainer>,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
