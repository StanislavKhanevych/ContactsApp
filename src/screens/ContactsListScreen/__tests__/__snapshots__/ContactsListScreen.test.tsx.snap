// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsListScreen should match the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    onPress={[Function]}
    selectable={false}
    style={
      [
        {
          "color": "#007AFF",
          "fontSize": 62,
        },
        {
          "bottom": 20,
          "position": "absolute",
          "right": 20,
          "zIndex": 1,
        },
        {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="add-contact"
  >
    î¨…
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search"
    style={
      {
        "backgroundColor": "#f1f1f1",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 0.5,
        "borderRadius": 10,
        "fontSize": 16,
        "marginHorizontal": 10,
        "marginVertical": 10,
        "paddingHorizontal": 15,
        "paddingVertical": 8,
      }
    }
    testID="search-bar"
    value=""
  />
  <RCTScrollView
    data={
      [
        {
          "data": [
            {
              "emailAddresses": [
                {
                  "email": "carl-jung@example.com",
                  "label": "work",
                },
              ],
              "familyName": "Jung",
              "givenName": "Carl",
              "hasThumbnail": true,
              "phoneNumbers": [
                {
                  "label": "mobile",
                  "number": "(555) 555-5555",
                },
              ],
              "recordID": "6b2237ee0df85980",
              "thumbnailPath": "content://com.android.contacts/display_photo/3",
            },
          ],
          "title": "C",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={
      [
        0,
      ]
    }
    testID="flat-list"
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "backgroundColor": "white",
              "borderBottomColor": "#f1f1f1",
              "borderBottomWidth": 1,
              "paddingHorizontal": 15,
              "paddingVertical": 5,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            C
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              {
                "borderBottomColor": "#f1f1f1",
                "borderBottomWidth": 1,
                "margin": 5,
                "paddingLeft": 10,
                "paddingRight": 70,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                  }
                }
              >
                Carl 
              </Text>
              <Text
                style={
                  {
                    "fontSize": 18,
                  }
                }
              >
                Jung
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
